crates/proto/src/rr/domain/name.rs

```diff
@@ -48,11 +51,37 @@ impl Name {
 
     /// Extend the name with the offered label, and ensure maximum name length is not exceeded.
     fn extend_name(&mut self, label: &[u8]) -> Result<(), ProtoError> {
+        let new_len = self.len() + label.len() + 1;
+
+        if new_len > Self::MAX_LENGTH {
+            return Err(ProtoErrorKind::DomainNameTooLong(new_len).into());
+        };
+
         self.label_data.extend_from_slice(label);
         self.label_ends.push(self.label_data.len() as u8);
-        if self.len() > 255 {
-            return Err(ProtoErrorKind::DomainNameTooLong(self.len()).into());
+
+        Ok(())
+    }
+
+    /// Prepend the name with the offered label, and ensure maximum name length is not exceeded.
+    fn prepend_name(&mut self, label: &[u8]) -> Result<(), ProtoError> {
```

Comment:
I'm not sure I like the name of this method as this is prepending a label, not the name. Additionally, I'm not sure the value this provides over the existing prepend_label which is generic over IntoLabel.

IntoLabel already has an implementation for `&[u8]`

```
impl [IntoLabel](https://docs.rs/hickory-proto/0.25.0-alpha.5/hickory_proto/rr/domain/trait.IntoLabel.html) for &[[u8](https://doc.rust-lang.org/nightly/std/primitive.u8.html)]
```

maybe you could explain why you think this is necessary? I feel like your new logic here could be put into the existing method, but maybe I've overlooked something.